ITEvents/Screens/Favorive tab/FavoritePresenter.swift

```diff
@@ -4,31 +4,41 @@ class FavoritePresenter: IFavoritePresenter {
     var view: IFavoriveView!
     var eventDataService: IEventsDataService!
     var selectedEventService: ISelectedEventService!
-    private var isListLayoutSelected = true
+    var userSettingsService: IUserSettingsService!
+    private var userSettings: UserSettings!
     
     init(view: IFavoriveView,
          eventDataService: IEventsDataService,
-         selectedEventService: ISelectedEventService) {
+         selectedEventService: ISelectedEventService,
+         userSettingsService: IUserSettingsService) {
         self.view = view
         self.eventDataService = eventDataService
         self.selectedEventService = selectedEventService
+        self.userSettingsService = userSettingsService
     }
     
     func setup(then completion: @escaping () -> Void) {
+        userSettingsService.fetchSettings { settings in
+            self.userSettings = settings
+        }
         eventDataService.fetchEvents { fetchedEvents in
             self.view.setEvents(fetchedEvents)
             completion()
         }
-        view.toggleListLayout(to: isListLayoutSelected)
+        view.toggleListLayout(to: userSettings.isListLayoutSelected)
     }
     
     func toggleLayoutState() {
-        isListLayoutSelected = !isListLayoutSelected
-        view.toggleListLayout(to: isListLayoutSelected)
+        userSettings.isListLayoutSelected = !userSettings.isListLayoutSelected
+        view.toggleListLayout(to: userSettings.isListLayoutSelected)
     }
     
     func selectEvent(_ event: Event) {
         selectedEventService.selectedEvent = event
     }
     
+    func saveStateBeforeDisappear(isListLayoutSelected: Bool) {
```

Comment:
it'd be better to have IFavoritePresenter.dispose (or IFavoritePresenter.destroy) whatever you call it to let the 
presenter handle all required stuff.
```swift
ViewController {
   override func viewWillDisappear(_ animated: Bool) {
       super.viewWillDisappear(animated)
       // here the view is even more stupid - knows nothing :) and we like it and we need it.
       presenter.destroy()
    }
}

FavoritePresenter {
   func destroy() {
       // userSettings should be already up-to-date see - func toggleLayoutState()
       userSettingsService.save(settings: userSettings)
    }
}
```